src/jit/lsraxarch.cpp

```diff
@@ -2287,10 +2287,7 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
     HWIntrinsicCategory category    = HWIntrinsicInfo::lookupCategory(intrinsicId);
     int                 numArgs     = HWIntrinsicInfo::lookupNumArgs(intrinsicTree);
 
-    if ((isa == InstructionSet_AVX) || (isa == InstructionSet_AVX2))
-    {
-        SetContainsAVXFlags(true, 32);
-    }
+    SetContainsAVXFlags(intrinsicTree->isSIMD(), intrinsicTree->gtSIMDSize);
```

Comment:
@CarolEidt, Do you know why the first parameter is `isFloatingPointType`?

I would think we just want to call `SetContainsAVX` when `canUseVexEncoding` is true for any node that generates a VEX-encoded instruction (and not just for floating-point types).

I would then think we also want to call `SetContains256BitAvx` for any node which generates a 256-bit AVX instruction....

So I would think this should be (which would require changing https://github.com/dotnet/coreclr/blob/master/src/jit/lsraxarch.cpp#L2962):
```cpp
if (intrinsicTree->IsSIMD())
{
    SetContainsAvxFlags(intrinsicTree->gtSimdSize);
}
```

Reply:
I suspect that this was first added when the only VEX encoded instructions we were using were for SIMD floating point types.

I think that your proposal is more correct than what we have now.

Reply:
@fiigii - it isn't immediately obvious why this is more correct. I think a comment would be useful saying something like "Set the AVX Flags if this instruction may use VEX encoding. Note that this may be true even if the ISA is not AVX (e.g. for platform-agnostic intrinsics or non-AVX intrinsics that will use VEX encoding if it is available on the target)."

Does that make sense?

Reply:
It makes sense, will change.